packages/playground/__tests__/Header.test.tsx

```diff
@@ -0,0 +1,139 @@
+import { Header } from '~/app/Header';
+import { render, screen, waitFor } from '@testing-library/react';
+import { AppContext, AppProvider } from '~/context';
+
+describe('Given the Header component is rendered', () => {
+  beforeEach(() => {
+    render(<Header />);
+  });
+
+  test('When it appears on the screen', () => {
+    const settingsButton = screen.getByText('Settings');
+    const logoIcon = screen.getByTestId('headerLogo');
+
+    // Then ...
+    expect(settingsButton).toBeInTheDocument();
+    expect(logoIcon).toBeInTheDocument();
+  });
+
+  test('When settings button was not clicked', () => {
+    const darkMode = screen.queryByText('Dark Mode');
+
+    // Then ...
+    expect(darkMode).toBeNull();
+  });
+
+  test('When settings button is clicked', () => {
+    const settingsButton = screen.getByText('Settings');
+    settingsButton.click();
+    const darkMode = screen.getByText('Dark Mode');
+    const minimap = screen.getByText('Minimap');
+    const numbering = screen.getByText('Numbering');
+
+    // Then ...
+    expect(darkMode).toBeInTheDocument();
+    expect(minimap).toBeInTheDocument();
+    expect(numbering).toBeInTheDocument();
+  });
+
+  test('When settings dropdown is closed', async () => {
+    const settingsButton = screen.getByText('Settings');
+    settingsButton.click();
+    const closeButton = screen.getByLabelText('close');
+    closeButton.click();
+
+    // Then ...
+    await waitFor(() => {
+      expect(screen.queryByText('Dark Mode')).toBeNull();
+    });
+  });
+
+  test('When settings button is clicked twice', async () => {
+    const settingsButton = screen.getByText('Settings');
+    settingsButton.click();
+    settingsButton.click();
+
+    // Then ...
+    await waitFor(() => {
+      expect(screen.queryByText('Dark Mode')).toBeNull();
+    });
+  });
+
+  test('When defaults are unchanged', () => {
```

Comment:
The tests above don't call render because it's done `beforeEach`. The tests below render something. Is this intended?

Reply:
Yes, I'd rather test the Menu and Toggle from the header rendered in beforeEach, like in the commented out test below, but there is this bug preventing clicks on the dropdown without throwing an error at the moment, eventhough the functionality works.

For testing the defaults in context, a consumer or components that consumes the context has to be rendered, because the "useContext" hook can only be called inside the body of a function component.

I would switch to an easier setup, as soon as the bugfix in primereact is released.